cmd/root.go

```diff
@@ -271,27 +271,43 @@ func (c *rootCommand) setupLoggers() (<-chan struct{}, error) {
 	}
 
 	loggerForceColors := false // disable color by default
-	switch c.logOutput {
-	case "stderr":
+	switch line := c.logOutput; {
+	case line == "stderr":
 		loggerForceColors = !noColor && stderrTTY
 		c.logger.SetOutput(stderr)
-	case "stdout":
+
+	case line == "stdout":
 		loggerForceColors = !noColor && stdoutTTY
 		c.logger.SetOutput(stdout)
-	case "none":
+
+	case line == "none":
 		c.logger.SetOutput(ioutil.Discard)
-	default:
-		if !strings.HasPrefix(c.logOutput, "loki") {
-			return nil, fmt.Errorf("unsupported log output `%s`", c.logOutput)
-		}
+
+	case strings.HasPrefix(line, "loki"):
 		ch = make(chan struct{})
-		hook, err := log.LokiFromConfigLine(c.ctx, c.fallbackLogger, c.logOutput, ch)
+		hook, err := log.LokiFromConfigLine(c.ctx, c.fallbackLogger, line, ch)
 		if err != nil {
 			return nil, err
 		}
 		c.logger.AddHook(hook)
 		c.logger.SetOutput(ioutil.Discard) // don't output to anywhere else
 		c.logFmt = "raw"
+
+	case strings.HasPrefix(line, "file"):
+		if c.fallbackLogger == nil {
+			return nil, fmt.Errorf("fallback logger is nil")
+		}
+
```

Comment:
Was this added because in some test this was true? 

I think this is currently not possible to happen in the real code, so I am against there being an error for it ;) (and if it can happen here it needs to be added to the `loki case as well`

Reply:
@MStoykov This was mostly suggested by me. It should make easier the detection of eventual nil logger in any possible future change. It would be difficult to detect an issue just by testing and/or running some examples because it would be used only in case of an error of the main writer. And in the case, the panic would happen we are losing the main error generated by the writer. That is my thought about why I requested to have it. WDYT?